wordplay.py

```diff
@@ -0,0 +1,244 @@
+        words_no_vowels_includingY.append(word)
+# To display answer as an array
+#print(vowel_in_word)
+
+#Question: 
+# What are all of the words that have all 5 vowels, in any order?
+vowels = "AEIOU"
+words_with_all_vowels = [] 
+for word in words: 
+    vowel_in_word = True
+    for char in vowels:
+        if char not in word:
+            vowel_in_word = False
+            break
+    if vowel_in_word:
+        words_with_all_vowels.append(word)
+# To Display Answer
+# print(words_with_all_vowels)
+
+#Question: 
+# What are all of the words that have all 5 vowels, in alphabetical order? 
+words_with_alphabetical_vowels = []
+vowels = "AEIOU"
+
+for word in words:
+    word_index = 0
+    vowel_index = 0
+    while word_index < len(word) and vowel_index < len(vowels):
+        if word[word_index] == vowels[vowel_index]:
+            vowel_index += 1
+        word_index += 1
+    if vowel_index == len(vowels):
+            words_with_alphabetical_vowels.append(word)
+#to display answer:
+#print(words_with_alphabetical_vowels)
+
+
+#Question: 
+#How many words contain the substring "TYPE”?
+type_count = 0
+for word in words:
+    if "TYPE" in word:
+        type_count += 1
+# To display an answer
+# print(type_count)
+
+#Question: 
+#Create and print an array containing all of the words that end in "GHTLY"
+end_in_GHTYL = []
+for word in words:
+    if word[-5:] ==  "GHTLY":
+         end_in_GHTYL.append(word)
+#To display all the words
+#print(end_in_GHTYL)
+
+#Question:
+#What is the shortest word that contains all 5 vowels?
+
+shortest_word_all_vowels = words_with_all_vowels[0]
+for word in words_with_all_vowels:
+    if len(word) < len(shortest_word_all_vowels):
+        shortest_word_all_vowels = word
+# To display answer:
+#print(shortest_word)
+
+
+
+#What is the longest word that contains no vowels? Assumes Y is allowed. 
+
+vowels = "AEIOU"
+words_no_vowels = []
+for word in words:
+    vowel_in_word = False
+    for char in vowels:
+        if char in word:
+            vowel_in_word = True
+            break 
+    if not vowel_in_word:
+        words_no_vowels.append(word)
+
+longest_word_no_vowels = words_no_vowels[0]
+for word in words_no_vowels:
+    if len(word) > len(longest_word_no_vowels):
+        longest_word_no_vowels = word
+
+#To display Answer
+#print(longest_word)
+    
+
+#Question
+#Which of the letters Q, X, and Z is the least common?
+letter_Q_count = 0
+letter_X_count = 0
+letter_Z_count = 0
+
+for word in words:
+    for char in word:
+        if "Q" == char:
+            letter_Q_count += 1
+        if "X" == char:
+            letter_X_count += 1
+        if "Z" == char:
+            letter_Z_count += 1
+#To Display Answer: 
+# if letter_Q_count and letter_X_count > letter_Z_count:
+#     print('letter Z count:', letter_Z_count)
+# elif letter_Z_count and letter_X_count > letter_Q_count:
+#     print('letter Q count:', letter_Q_count)
+# else:
+#     print('letter X count:', letter_X_count)
```

Comment:
I think there's a correctness issue in your evaluation code here:

```python
letter_Q_count = 1
letter_X_count = 100
letter_Z_count = 50

if letter_Q_count and letter_X_count > letter_Z_count:
  print("Uh oh") # This gets printed
```

Can you spot the bug here?

It's also present on the `elif` statement

Reply:
Hmm I think so... my intention was to see if Z has the lowest count, but my code doesn't do this. Instead the if statement checks if letter_Q_count is a truthy value (not 0) and if letter_X_count is higher than letter_Z_count. I need to change it to letter_Q_count > letter_Z_count and letter_X_count > letter_Z_count to do what I want here. Nice catch, thank you

Reply:
Yep - that's the issue exactly, well stated